temporalio/common.py

```diff
@@ -35,6 +37,21 @@ class RetryPolicy:
     non_retryable_error_types: Optional[Iterable[str]] = None
     """List of error types that are not retryable."""
 
+    @staticmethod
```

Comment:
Other SDKs have some client side validation for some of these values: https://github.com/temporalio/sdk-typescript/blob/e147ad5ddbc2f2446dc65064cf512b29a8ce0a1f/packages/internal-workflow-common/src/retry-policy.ts#L48

Reply:
In Go I think we only have it for the unit test environment and it relies on server validation (specifically mentions at https://github.com/temporalio/sdk-go/blob/cee305f568f4b7cba5ec1c597d206bd0114289d7/internal/internal_workflow_testsuite.go#L1230-L1258 they copied server validation). But I can add it here.

Also, that doesn't catch things like negative initial interval that server does (and I of course matching server's validation in general has issues since they may change it).

Reply:
I think we have the same in Java.
I don't remember why I added it in TS, there was some issue that required that we do some basic validation on the SDK.

Reply:
It's harmless to do the obvious validation client side, so I'm adding it

Reply:
Done